### DISCUSSION TITLE
How to understand the third considerations of Reactivity docs?

### DISCUSSION THREAD

**Discussion by mapinxue** [solidjs/solid#1271] [Q&A] [answered]
In the docs about `solid reactivity`, I see the third consideration:
> This approach only tracks synchronously. If you have a setTimeout or use an async function in your Effect the code that executes async after the fact won't be tracked.  --- [docs link](https://www.solidjs.com/guides/reactivity#considerations)

But I can not create the situation as that. Besides, I found it runs correctly  in [the first example(counter)](https://www.solidjs.com/examples/counter). That really make me confused. Can somebody just give me a simple example of this? sincerely appreciate.

## Screen shot
<img width="959" alt="image" src="https://user-images.githubusercontent.com/73047992/195378897-99c0b596-6858-4274-ac79-2e1f04e9c57e.png">

**Comment by ryansolid** [comment, marked as answer]
It means that it cannot track in an async context. You can always write to reactive values async but it won't track in async functions in a computation.

```js
const [count, setCount] = createSignal(0);

// console.log our count
createEffect(( ) => {
  // this logs every time count updates
  console.log(count())
})

// let's delay logging for a second
createEffect(( ) => {
  // this will only log the first time as count is read in an async function
  setTimeout(() => console.log(count()), 1000)
})

// let's delay logging for a second
createEffect(( ) => {
  // this will work as count is read synchronously
  const value = count();
  setTimeout(() => console.log(value), 1000)
})
```

**Reply by mapinxue** [reply]
awesome. I added a line to the last line to maybe make it more clear.
```js
setTimeout(() => setCount(1), 2000);
```
then, the first and third effect react. thanks a lot.
